### PR TITLE
arduino_nano_33_ble: enable the adafruit 2.8 TFT display

### PR DESCRIPTION
**Pull request by brgl**
The following device-tree changes make it possible to connect the adafruit 2.8 TFT display with patch wires or a hand-made adapter to the Arduino Nano 33 BLE and make the display and touchscreen work.

The following command builds an lvgl Hello World image: `west build  -b arduino_nano_33_ble ./samples/subsys/display/lvgl/ -- -DSHIELD=adafruit_2_8_tft_touch_v2`.

Normally the display is connected using the Rev3 Arduino header but the nano form factor doesn't have one. This doesn't mean the display can't be used however as it only needs SPI and I2C (and two GPIOs) hence the PR.

### CODE REVIEW

File: boards/shields/adafruit_2_8_tft_touch_v2/boards/arduino_nano_33_ble.overlay [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 Huawei Inc.
```
Comment (erwango) [inline boards/shields/adafruit_2_8_tft_touch_v2/boards/arduino_nano_33_ble.overlay:2]: Please keep usual project headers:
```
/*
 * Copyright (c) 2016 Open-RnD Sp. z o.o.
 *
 * SPDX-License-Identifier: Apache-2.0
 */
```

**Review by erwango — CHANGES_REQUESTED** [review]
One requst otherwise LGTM

**Review by carlocaione — DISMISSED** [review]

File: boards/shields/adafruit_2_8_tft_touch_v2/boards/arduino_nano_33_ble.overlay (hunk 1 above)
Comment (brgl) [reply boards/shields/adafruit_2_8_tft_touch_v2/boards/arduino_nano_33_ble.overlay]: Done.

**Review by carlocaione — APPROVED** [review]

**Comment by brgl** [comment]
This one is pretty straightforward - can it be merged now? @carlescufi? @MaureenHelm?

**Review by erwango — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by jfischer-no** [comment]
> This one is pretty straightforward - can it be merged now? @carlescufi? @MaureenHelm?

This is blatantly wrong, arduino_nano_33_ble board has an Arduino Nano header, there is even a PR that introduces the Arduino Nano connector, https://github.com/zephyrproject-rtos/zephyr/pull/41961. Further, how someone connects an improper shield (Arduino connector) to a board (Arduino Nano connector) is their own problem and has nothing to do with the project.
@brgl Please revert asap.

**Comment by brgl** [comment]
@jfischer-no I wouldn't disagree if the changes had any impact on introducing the arduino nano header you mentioned but they don't. The dummy header definition is only included if you build the shield overlay and doesn't break any existing setups.

As to your second point: there are adapters for arduino nano to Rev3. I don't see why you would want to make it harder for people to use a popular display (by not letting it be upstreamed for a particular arch). If it can be made to work with an adapter or a bunch of wires and doesn't break any existing or future setup - there's no reason to remove it.

**Comment by jfischer-no** [comment]
> @jfischer-no I wouldn't disagree if the changes had any impact on introducing the arduino nano header you mentioned but they don't. The dummy header definition is only included if you build the shield overlay and doesn't break any existing setups.

Your first commit is wrong and after #41961 the node labels must be changed to `arduino_nano_*`. This also makes your second commit void.

> As to your second point: there are adapters for arduino nano to Rev3. I don't see why you would want to make it harder for people to use a popular display (by not letting it be upstreamed for a particular arch). If it can be made to work with an adapter or a bunch of wires and doesn't break any existing or future setup - there's no reason to remove it.

One can do it right, with #41961 one can extend this display shield generically so that _popular_ display works with any Arduino Nano board. What you have done in the second commit is short-sighted and simply wrong.

**Comment by brgl** [comment]
Fair enough. As it's not blocking anyone, I prefer to look into it and prepare a follow up that'll also convert the 33 BLE version to those new labels.

**Comment by carlescufi** [comment]
> Fair enough. As it's not blocking anyone, I prefer to look into it and prepare a follow up that'll also convert the 33 BLE version to those new labels.

Thank you @brgl please do post a follow-up with the changes requested by @jfischer-no.

**Pull request merged**